metal/modules/lstm_module.py

```diff
@@ -137,29 +146,40 @@ def forward(self, X):
         """Applies one step of an lstm (plus reduction) to the input X
 
         Args:
-            X: (torch.LongTensor) of shape [batch_size, max_seq_length].
-                The indices of the embeddings to look up for each item in the
-                batch.
+            X: (torch.LongTensor) of shape [batch_size, max_seq_length],
+            containing the indices of the embeddings to look up for each item in
+            the batch; OR, if self.skip_embeddings == True and features are
+            directly being passed in, then has shape [batch_size,
+            max_seq_length, feat_dim].
         """
+        # Check that X has the correct format
+        if len(X.shape) == 2 and not self.skip_embeddings:
+            X = X.long()
+        elif len(X.shape) == 3 and self.skip_embeddings:
+            X = X.float()
+        else:
+            raise ValueError(f"X {X.shape} and skip_embeddings do not match.")
```

Comment:
Why different types here based on dimensionality?  Seems a bit odd...maybe add comment or find a way to make this a bit more compact?

Reply:
This is just handling the two different types as specified in the docstring- think this is okay... alternative would be a sub-class but I think that's overkill here